mayastor/src/bdev/dev/nvmx/channel.rs

```diff
@@ -226,24 +226,36 @@ impl NvmeControllerIoChannel {
 
         debug!("Creating IO channel for controller ID 0x{:X}", id);
 
-        let controller = match NVME_CONTROLLERS.lookup_by_name(id.to_string()) {
+        let carc = match NVME_CONTROLLERS.lookup_by_name(id.to_string()) {
             None => {
                 error!("No NVMe controller found for ID 0x{:X}", id);
                 return 1;
             }
             Some(c) => c,
         };
 
-        let controller = controller.lock().expect("lock error");
-
-        // Make sure controller is available.
-        if controller.get_state() != NvmeControllerState::Running {
-            error!(
-                "{} controller is in {:?} state, I/O channel creation not possible",
-                controller.get_name(),
-                controller.get_state()
-            );
-            return 1;
+        let cname;
+        let spdk_handle;
+
+        {
```

Comment:
Not an issue, but I'd note Rust is fine with multiple returns via a tuple:

```suggestion
        let (cname, spdk_handle) = {
```

Then you can later return:

```rust
(controller.get_name(), controller.ctrlr_as_ptr())
```

Reply:
Thanks for the suggestion, the code looks nicer. Fixed.